src/ddsrt/include/dds/ddsrt/hopscotch.h

```diff
@@ -81,29 +226,194 @@ struct ddsrt_chh_iter {
 };
 #endif
 
+/**
+ * @brief Concurrent version of @ref ddsrt_hh_new
+ * 
+ * @param[in] init_size the minimum initial size
+ * @param[in] hash the hash function, see @ref ddsrt_hh_hash_fn
+ * @param[in] equals the equals function, see @ref ddsrt_hh_equals_fn
+ * @param[in] gc_buckets user defined garbage collection function, see @ref ddsrt_hh_buckets_gc_fn
+ * @param[in] gc_buckets_arg extra argument for @ref ddsrt_hh_buckets_gc_fn
+ * @return pointer to the hopscotch hash table
+ * @see @ref ddsrt_chh_free
+ */
 struct ddsrt_chh *ddsrt_chh_new (uint32_t init_size, ddsrt_hh_hash_fn hash, ddsrt_hh_equals_fn equals, ddsrt_hh_buckets_gc_fn gc_buckets, void *gc_buckets_arg);
+
+/**
+ * @brief Concurrent version of @ref ddsrt_hh_free
+ * 
+ * @param[in,out] hh the hash table to destroy
+ * @see @ref ddsrt_chh_new
+ */
 void ddsrt_chh_free (struct ddsrt_chh * __restrict hh);
+
+/**
+ * @brief Concurrent version of @ref ddsrt_hh_lookup
```

Comment:
I think it would be worth pointing out that a lookup is lock- and wait-free and can occur in parallel to a remove operation, and that a successful lookup doesn't mean the returned element is still part of the hash table by the time control has been returned to the caller.

Reply:
As a further feedback I'd say you might want to provide a return value indicating why it failed, since there are two options: 

1. It failed because the element doesn't exist
2. It failed because another thread was busy with it for too long.

I think it would be wrong for the program to assume 1 upon failure when it actually failed due to 2, since in the latter case it might decide to retry at some later time and still get it.